apps/api/src/modules/job/job.service.spec.ts

```diff
@@ -1,19 +1,71 @@
 import { Test, TestingModule } from '@nestjs/testing'
+import { Job } from '@prisma/client'
+import { mock } from 'src/common/mocks'
+import { PrismaService } from 'src/database/prisma.service'
 
+import { GetJobCardWithMaxPageDto, SearchJobDto } from './job.dto'
+import { JobRepository } from './job.repository'
 import { JobService } from './job.service'
 
 describe('JobService', () => {
   let service: JobService
+  let repository: JobRepository
 
   beforeEach(async () => {
+    //test module
     const module: TestingModule = await Test.createTestingModule({
-      providers: [JobService],
+      providers: [JobService, JobRepository, PrismaService],
     }).compile()
 
     service = module.get<JobService>(JobService)
+    repository = module.get<JobRepository>(JobRepository)
   })
 
   it('should be defined', () => {
     expect(service).toBeDefined()
   })
+  it('should get all job successfully', async () => {
+    const limitQueryMax = 10 //loop check all limitQuery 1-10
+    const jobDataLength = 10 //fixed mock data length in repository
+
+    //mock repository
+    jest
+      .spyOn(repository, 'getJob')
+      .mockResolvedValue(mock('job').get(jobDataLength))
+
+    //get mocked repository
+    const itMockedJobData = await repository.getJobJoined({
+      take: jobDataLength,
+    })
```

Comment:
Don't use repository to get data in test, maybe use this instead
```suggestion
    const itMockedJobData:GetJobCardDto[] = Array.from({length: jobDataLength}, (v,i)=>({
      jobId: i+1,
      ...mock('job').omit(['jobId']).get(),
      ...mock('casting').pick(['companyName']).get(),
      ...mock('user').pick(['profileImageUrl']).get(),
    }))
```